src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -432,7 +432,7 @@ public Activity AddBaggage(string key, string? value)
         {
             KeyValuePair<string, string?> kvp = new KeyValuePair<string, string?>(key, value);
 
-            if (_baggage != null || Interlocked.CompareExchange(ref _baggage, new LinkedList<KeyValuePair<string, string?>>(kvp), null) != null)
+            if (_baggage != null || Interlocked.CompareExchange(ref _baggage, new LinkedList<KeyValuePair<string, string?>>(kvp, reversed: true), null) != null)
```

Comment:
I do think this works as written, but what would you think of creating a LinkedList.AddLast() API and invoking that here rather than storing extra state in the LinkedList? To me that would be a little bit simpler and has some minor performance benefits: 4-8 bytes less memory used in the LinkedList object and avoids adding a new conditional branch in the code path for tags, events, and baggage.